### PR TITLE
[LETS-134] Add system parameter to dump fileio cache

### PR DESCRIPTION
**Pull request by adrianbrab**
http://jira.cubrid.org/browse/LETS-134

Added a new hidden system parameter 'dump_fileio_cache_after_boot' used to dump cached volume information at the end of boot.
This a boolean parameter with the default value set on false.

Example of parameter output:
```
Time: 07/06/21 12:15:54.945 - DEBUG *** file /home/adrian/cubridScalabil/cubrid/src/storage/file_io.c, line 2123
Cached volume with volid=-2, vdes=110, label=/home/adrian/cubridScalabil/db/testdb/testdb_lgat

Time: 07/06/21 12:15:54.945 - DEBUG *** file /home/adrian/cubridScalabil/cubrid/src/storage/file_io.c, line 2123
Cached volume with volid=-21, vdes=113, label=/home/adrian/cubridScalabil/db/testdb/testdb_lgar_t

Time: 07/06/21 12:15:54.945 - DEBUG *** file /home/adrian/cubridScalabil/cubrid/src/storage/file_io.c, line 2123
Cached volume with volid=0, vdes=111, label=/home/adrian/cubridScalabil/db/testdb/testdb
```

### CODE REVIEW

File: src/storage/file_io.c [hunk 1]
```diff
@@ -502,6 +502,10 @@ static bool fileio_synchronize_sys_volume (THREAD_ENTRY * thread_p, FILEIO_SYSTE
 static bool fileio_synchronize_volume (THREAD_ENTRY * thread_p, FILEIO_VOLUME_INFO * vol_info_p, APPLY_ARG * arg);
 static int fileio_cache (VOLID volid, const char *vlabel, int vdes, FILEIO_LOCKF_TYPE lockf_type);
 static void fileio_decache (THREAD_ENTRY * thread_p, int vdes);
+void fileio_cache_dump (void);
+static void fileio_dump_cached_file (VOLID volid, int vdes, const char *vlabel);
```
Comment (ribram) [inline src/storage/file_io.c]: Already declared in the header.
```suggestion
static void fileio_dump_cached_file (VOLID volid, int vdes, const char *vlabel);
```

File: src/transaction/boot_sr.c [hunk 2]
```diff
@@ -2385,6 +2385,11 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
 	}
     }
 
+//  if (prm_get_bool_value (PRM_ID_DUMP_FILE_CACHE))
+//    {
+//      fileio_cache_dump (thread_p);
+//    }
+
```
Comment (ribram) [inline src/transaction/boot_sr.c]:
```suggestion
```

Comment (ribram) [reply src/transaction/boot_sr.c]: already resolved

File: src/storage/file_io.c (hunk 1 above)
Comment (adrianbrab) [reply src/storage/file_io.c]: Removed now

**Review by ribram — APPROVED** [review]

File: src/storage/file_io.c [hunk 3]
```diff
@@ -2091,6 +2094,60 @@ fileio_close (int vol_fd)
...
+}
+
+/*
+ * fileio_dump_sysvolinfo () - Iterator function to call fileio_dump_volinfo
+ * to print system volume information.
+ *    return: false to keep the iteration going
+ *    sys_volinfo(in): System volume information structure
+ */
+static bool
+fileio_dump_sysvolinfo (THREAD_ENTRY *, FILEIO_SYSTEM_VOLUME_INFO * sys_volinfo, APPLY_ARG *)
```
Comment (cristiarg) [inline src/storage/file_io.c:2145]: I checked and the last param could be changed to `const APPLY_ARG *` as nothing modifies it. But maybe this is out of scope as it's a crosscutting change.

**Review by cristiarg — APPROVED** [review]

File: src/storage/file_io.c (hunk 3 above)
Comment (adrianbrab) [reply src/storage/file_io.c:2145]: I think is better to leave it as nonconst for now as it will fit better with the template of the other apply functions.

Comment (ribram) [reply src/storage/file_io.c:2145]: I think @cristiarg suggestion is to update all the apply functions. But this is indeed very out of scope :).
